[WIP] Added Delfin API support to dashboard.

anvithks opened the pull request:
**What type of PR is this?**
/kind new feature

**What this PR does / why we need it**:
This PR adds the support for Delfin APIs to the dashboard.

**Which issue(s) this PR fixes**:
Fixes #370 #84 

**Test Report Added?**:

> /kind TESTED
/kind NOT-TESTED

**Test Report**:


**Special notes for your reviewer**:

NajmudheenCT commented on entrypoint.sh:
```diff
@@ -16,12 +16,14 @@
 
 OPENSDS_HOTPOT_URL=${OPENSDS_HOTPOT_URL:-http://127.0.0.1:50040}
 OPENSDS_ORCHESTRATION_URL=${OPENSDS_ORCHESTRATION_URL:-http://127.0.0.1:5000}
+SODA_DELFIN_URL=${SODA_DELFIN_URL:-http://127.0.0.1:8190}
```
Is this a configurable (user given ) setting but defaults to 8190 ?

anvithks replied:
This is the configuration file for the installation. If the Port is going to be dynamic then it will have to be set in the installation steps and read here during the setup.

NajmudheenCT commented on entrypoint.sh:
```diff
@@ -16,12 +16,14 @@
 
 OPENSDS_HOTPOT_API_VERSION=${OPENSDS_HOTPOT_API_VERSION:-v1beta}
 OPENSDS_ORCHESTRATION_API_VERSION=${OPENSDS_ORCHESTRATION_API_VERSION:-orch}
+SODA_DELFIN_API_VERSION=${SODA_DELFIN_API_VERSION:-delfin}
```
mention delfin here is for version default value? if so you can default it to v1

anvithks replied:
This is a dashboard specific name resolution. Since the API version is same for Gelato and Delfin `v1` the dashboard code needs to be routed to the appropriate API. This is where the name comes in. the naming needs to be updated as they are not really API versions. I will make the appropriate changes

NajmudheenCT commented on entrypoint.sh:
```diff
@@ -45,6 +47,10 @@ cat > /etc/nginx/conf.d/default.conf <<EOF
         }
+
+         location /${SODA_DELFIN_API_VERSION}/ {
+            proxy_pass ${SODA_DELFIN_URL}/${OPENSDS_GELATO_API_VERSION}/;
```
please help to understand , why GELATO version here?

anvithks replied:
Since Delfin is using V1 just reused the Gelato version. This will be changed in final PR

NajmudheenCT commented on proxy.conf.json:
```diff
@@ -19,6 +19,15 @@
         "logLevel" : "debug"
     },
+    "/delfin" : {
+        "target": "http://192.168.56.110:8190",
```
is this a sample deafult cof file ? target can be 127.0.0.0 ??

anvithks replied:
This is local development prox conf. The IP address has to be replaced with the local address. This info is in the development branch Readme.

anvithks commented:
> Please add some screenshots.. which will help us to review html files

Currently the code that is in place is mostly to do with the routing and adding functionality and implementing the CRUD features and APIs. 
When the screens for the UI design are finalised and updated here I will start adding the screenshots.
This PR is still WIP so there will be quite a few changes. Suggest you wait for the changes before reviewing.

anvithks commented:
Updated the Register Storage page as per the updated API spec.

**Rest and SSH Config Disabled**
![register-storage-all-disabled](https://user-images.githubusercontent.com/19162717/91440382-26aa9380-e88c-11ea-9197-5bbc0a59ab22.png)

**Extra Attributes Enabled**
![register-storage-extra-attribs-enabled](https://user-images.githubusercontent.com/19162717/91440387-290ced80-e88c-11ea-8b4f-f97b9619f897.png)

**Rest Configuration Enabled**
![register-storage-rest-enabled](https://user-images.githubusercontent.com/19162717/91440391-29a58400-e88c-11ea-99a6-7ffe73ccd52d.png)

**Rest and SSH Configuration Enabled**
![register-storage-rest-ssh-enabled](https://user-images.githubusercontent.com/19162717/91440396-2ad6b100-e88c-11ea-9f39-dd28e6295d86.png)

**SSH Configuration Enabled**
![register-storage-ssh-enabled](https://user-images.githubusercontent.com/19162717/91440400-2b6f4780-e88c-11ea-8879-125ccdc72ff6.png)

**SSH Configuration Enabled with Pub Key Type**
![register-storage-ssh-enabled-with-pubkeyoptions](https://user-images.githubusercontent.com/19162717/91440404-2c07de00-e88c-11ea-9772-08534c6619e1.png)

anvithks commented:
Updated the PR. Added widgets and placeholders for the following: 
- List Storage Arrays by vendor / model
- Show top 5 Alerts
- Array Raw Capacity (WIP)
- Array usable Capacity (WIP)

![storage-summary-widget-1](https://user-images.githubusercontent.com/19162717/91896691-e77aa900-ecb6-11ea-8de1-ca141e60a889.png)
![storage-summary-widgets-2](https://user-images.githubusercontent.com/19162717/91896699-e9dd0300-ecb6-11ea-8d04-fc9b577e9229.png)

anvithks commented:
Closing this PR. Will re raise the PR against individual tasks / issues raised.
